midware/middleware.go

```diff
@@ -166,7 +175,15 @@ type traceRequestHeader struct {
 
 func (h *traceRequestHeader) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	var reqid string
+	precedenceHeaders := []string{DefaultTraceHeader, DefaultAzureHeader, DefaultAWSHeader}
+
 	if h.allow {
+		for _, header := range precedenceHeaders {
+			if r.Header.Get(header) != "" {
+				h.header = header
+				break
+			}
+		}
 		reqid = r.Header.Get(h.header)
```

Comment:
really minor nit, and feel free to not update the code if you don't agree: right now you're looking up r.Header.Get(header) twice, when you code save it as `reqid` and continue to override that variable.